File: apigateway_getting_started/how_to_encodings.py
# snippet-start:[serverless-snippets.python.how_to_encodings]
import base64
import random

import boto3

s3 = boto3.client("s3")


def lambda_handler(event, context):
    number = random.randint(0, 1)
    if number == 1:
        response = s3.get_object(
            Bucket="bucket-name",
            Key="image.png",
        )
        image = response["Body"].read()
        return {
            "headers": {"Content-Type": "image/png"},
            "statusCode": 200,
            "body": base64.b64encode(image).decode("utf-8"),
            "isBase64Encoded": True,
        }
    else:
        return {
            "headers": {"Content-type": "text/html"},
            "statusCode": 200,
            "body": "&lt;h1&gt;This is text&lt;/h1&gt;",
        }
# snippet-end:[serverless-snippets.python.how_to_encodings]

File: apigateway_getting_started/how_to_encodings.js
import { S3Client, GetObjectCommand } from "@aws-sdk/client-s3"

const client = new S3Client({});

export const handler = async (event) => {
    const randomint = function (max) {
        return Math.floor(Math.random() * max);
    }
    const number = randomint(2);
    if (number == 1) {
        const input = {
            "Bucket": "bucket-name",
            "Key": "image.png"
        }
        let base64body = "";
        try {
            const command = new GetObjectCommand(input)
            const response = await client.send(command);
            const bodyStr = await response.Body.transformToByteArray();
            base64body = Buffer.from(bodyStr).toString('base64');
        } catch (err) {
            console.error(err);
        }
        return {
            'headers': { "Content-Type": "image/png" },
            'statusCode': 200,
            'body': base64body,
            'isBase64Encoded': true
        }
    } else {
        return {
            'headers': { "Content-Type": "text/html" },
            'statusCode': 200,
            'body': "&lt;h1&gt;This is text&lt;/h1&gt;",
        }
    }
}